hummingbot/connector/connector_base.pyx

```diff
@@ -264,6 +265,47 @@ cdef class ConnectorBase(NetworkIterator):
         """
         raise NotImplementedError
 
+    def batch_order_create(self, orders_to_create: List[LimitOrder]) -> List[LimitOrder]:
```

Comment:
I think that we should allow also to create market orders too in batch, can we use OrderCandidate to do so?

Reply:
A while back @Nullably made the point that we don't want to ever use market orders. He showed a shallow order book where the first few ask levels were reasonably priced with small sizes, and the next one (say, third level) was a huge order to sell a large size at a very high price (say, if the mid-price was $1, this would be an ask for $10). So now we take the approach of always submitting limit orders, but if we want it to fill immediately, we just set the price to something like $12 and mark it as maker=False. This offers protection while still acting essentially as a market order

Reply:
Btw, note that the order candidate has a mandatory `price` field as well. It's for that reason — it always assumes a limit order.

Reply:
As discussed on Discord, I have pushed an update to this that will let us submit `MarketOrder`s as well. I've tested the change manually by adapting the `batch_order_update.py` strategy script.